src/utils/vc/document-loader.js

```diff
@@ -18,16 +18,22 @@ function documentLoader(resolver = null) {
    */
   async function loadDocument(uri) {
     let document;
-
-    if (resolver && uri.startsWith('did:')) {
+    let uriString = uri.toString();
+    if (resolver && uriString.startsWith('did:')) {
       // Try to resolve a DID and throw if cannot resolve
-      document = await resolver.resolve(uri);
+      document = await resolver.resolve(uriString);
     } else {
-      const cachedData = cachedUris.get(uri);
+      // Strip ending slash from uri
```

Comment:
well, this is only done for our caching system, so perhaps we can only strip it from the cache key to check instead then we dont have to worry about "is it safe". my initial feeling is it would be fine, but lets not open any doors and manipulate user input for remote requests. will make that change